File: CVcodes/canny.cpp
#include "opencv2/highgui/highgui.hpp"
#include "opencv2/imgproc/imgproc.hpp"
#include <stdio.h>

using namespace cv;
using namespace std;

int main(int argc, char* argv[])
{
    Mat img, gray, blur, canny_img;
    img = imread(argv[1]);
    cvtColor(img, gray, CV_BGR2GRAY);
    GaussianBlur(gray, blur, Size(3, 3), 0);

    int thresh = 100;
    Canny(blur, canny_img, thresh, thresh*2, 3);

    imshow("image", img);
    imshow("Canny", canny_img);

    waitKey(0);
    return 0;
}

File: CVcodes/canny.py
import cv2
import sys

img = cv2.imread(sys.argv[1])
gray = cv2.cvtColor(img, cv2.cv.CV_BGR2GRAY)
blur = cv2.GaussianBlur(gray, (3, 3), 0)

thresh=100

canny_img = cv2.Canny(blur, threshold1=thresh, 
                    threshold2=thresh*2, apertureSize=3)

cv2.imshow("image", img)
cv2.imshow("canny", canny_img)

cv2.waitKey(0)